test/main.js

```diff
@@ -62,38 +50,45 @@ describe('rebuildNativeModules', function() {
 
   const moduleVersionsToTest = ['0.22.0', '0.31.2'];
 
-  for(let nativeModuleVersionToBuildAgainst of moduleVersionsToTest) {
-    it(`Rebuilds native modules against ${nativeModuleVersionToBuildAgainst}`, async () => {
-      const targetHeaderDir = path.join(__dirname, 'testheaders');
-      const targetModulesDir = path.join(__dirname, 'node_modules');
-
+  for(let version of moduleVersionsToTest) {
+    it(`Rebuilds native modules against ${version}`, async () => {
       try {
-        if (await fs.stat(targetHeaderDir)) {
-          await rimraf(targetHeaderDir);
-        }
-      } catch (e) { }
-
-      await fs.mkdir(targetHeaderDir);
-
-      try {
-        if (await fs.stat(targetModulesDir)) {
-          await rimraf(targetModulesDir);
-        }
-      } catch (e) { }
-
-      await installNodeHeaders(nativeModuleVersionToBuildAgainst, null, targetHeaderDir);
-      let canary = await fs.stat(path.join(targetHeaderDir, '.node-gyp', nativeModuleVersionToBuildAgainst, 'common.gypi'));
-      expect(canary).to.be.ok
-
-      // Copy our own node_modules folder to a fixture so we don't trash it
-      await cp(path.resolve(__dirname, '..', 'node_modules'), targetModulesDir);
```

Comment:
@paulcbetts, the test as it was previously written doesn't test any output or byproduct of the rebuild.  It just executes the rebuild, cleans up, and exits.  Do think there is a value in providing an explicit before and after condition, then asserting against it?  I would think that using the node_modules/ folder puts the vendor modules in control of the prebuild state and postbuild state, as opposed to an explicit before-and-after state (in my case, the absence then presence of the file).  further, what's the value of testing all modules at once vs. one explicit one?